experimental/InjectiveResolutions/src/InjectiveResolutions.jl

```diff
@@ -0,0 +1,573 @@
+        S = sub(N,[h_N(g)]) #submodule of N =(0 :_M P_F)/(y0,...,yn) generated by g
+        if annihilator(S[1]) == PF && !is_zero(S[1])
+        #if is_subset(PF,annihilator(S[1]))
+            push!(B,g)
+        end
+        M_B,_ = sub(M,B)
+        N,h_N = quo(M,M_B)
+        if is_zero(N)
+            break
+        end
+    end
+    return filter(!is_zero,B) 
+end
+
+#Compute coefficients as in algorithm 3.5
+#INPUT:     SubquoModule M
+#           SubquoModule Mxy = (0 :_M P_F)
+#           homomorphism (inclusion) Mxy -> M
+#           k[ZF]-basis computed as in ZF_basis(_,_)
+function coefficients_incl(N::SubquoModule,Np::SubquoModule,incl_Np::SubQuoHom,Bp::Vector,pr::Ideal)
+    R_N = base_ring(N)
+    if is_zero(modulus(R_N))
+        R_poly = R_N
+    else
+        R_poly = base_ring(R_N)
+    end
+    T = typeof(zero(R_N))
+    lambda_out = Vector{Vector{T}}()
+    p_F = ideal_in_poly_ring(pr)
+    i = 1
+    for g in gens(N) 
+        m_g = monomial_basis(R_N,degree(g))[1]
+        N_g,incl_N_g = sub(N,[g;]) #submodule of M generated by g
+        I_g,incl_g,incl_p = intersect(Np,N_g) # (0 :_(g) PF)
+        lambda_g = Vector{T}()
+        j = 1
+        for b in Bp 
+            lambda_gb = R_N() #zero in R_M
+            if length(coordinates(ambient_representative(b))) > 1 #ugly workaround for the special case not considered in algorithm 3.6
+                a_yg = degree(b) - degree(g)
+                coeff_gb = coordinates(ambient_representative(b))[i] # coefficient of b on m_yg*g
+                if is_zero(coeff_gb)
+                    lambda_gb = 0*one(R_N)
+                else
+                    lambda_gb = monomial_basis(parent(coeff_gb),degree(coeff_gb)-a_yg)[1]*one(R_N)
+                end
+            else
+                for g_I in filter(!is_zero,gens(I_g)) 
+                    b_m = monomial_basis(R_N,degree(b))[1]
+                    g_I_m = monomial_basis(R_N,degree(g_I))[1]
+                    if equal_mod_ZF(b_m,g_I_m,p_F)
+                        a_yg = degree(b) - degree(g)
+                        m_yg = monomial_basis(R_poly,a_yg)[1] #monomial of degree a_yg
+                        gg = preimage(incl_Np,m_yg*g) # m_yg*g as an element of Np !not unique
+                        lambda_gb = m_g*coordinates(gg)[j] # coefficient of b on m_yg*g
+                    end
+                end
+            end
+            push!(lambda_g,lambda_gb) 
+            j = j + 1
+        end
+        push!(lambda_out,lambda_g)
+        i = i + 1
+    end
+    return matrix(R_N,lambda_out) 
+end
+
+#Returns ideal of underlying polynomial ring
+#INPUT:     ideal I of quotient ring
+#OUTPUT:    ideal I as an ideal of polynomial ring
+function ideal_in_poly_ring(I::Ideal)
+    R_I = base_ring(I)
+    J = modulus(R_I)
+    if is_zero(J)
+        return I
+    else
+        S = base_ring(R_I)
+        G = []
+        for g in gens(I) 
+            push!(G,monomial_basis(S,degree(g))[1])
+        end
```

Comment:
More idiomatic and also more type stable (i.e., the vector this produces not always a `Vector{Any}`) is to use a list comprehension:
```suggestion
        G = [monomial_basis(S,degree(g))[1] for g in gens(I)]
```